### ISSUE TITLE
Document that subprocess.Popen does not set PWD

### ISSUE THREAD

**Issue by 56441241-859c-4a4e-bf60-a8192e84927a** [python/cpython#74981]
BPO | [30798](https://bugs.python.org/issue30798)
--- | :---
Nosy | @bitdancer

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2017-06-29.02:32:34.277>
labels = ['type-feature', 'docs']
title = 'Document that subprocess.Popen does not set PWD'
updated_at = <Date 2017-07-02.00:38:27.753>
user = 'https://bugs.python.org/jamesdlin'
```

bugs.python.org fields:
```python
activity = <Date 2017-07-02.00:38:27.753>
actor = 'jamesdlin'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation']
creation = <Date 2017-06-29.02:32:34.277>
creator = 'jamesdlin'
dependencies = []
files = []
hgrepos = []
issue_num = 30798
keywords = []
message_count = 3.0
messages = ['297244', '297499', '297500']
nosy_count = 3.0
nosy_names = ['r.david.murray', 'docs@python', 'jamesdlin']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue30798'
versions = ['Python 3.6']
```

</p></details>

**Comment by 56441241-859c-4a4e-bf60-a8192e84927a** [comment]
Even though http://bugs.python.org/issue4057 was rejected (which I think is fair), I think it would be worth mentioning something about PWD in the Python docs for subprocess.Popen's cwd parameter (and possibly for os.chdir):

1. It's pretty common for people to use Python to execute other things.
2. I don't think it's that uncommon to come across binaries/scripts that directly read PWD instead of calling getcwd() or equivalent.
3. For people who encounter this problem for the first time, it can be really puzzling.

My suggestion would be to add something like:

   Note: Some programs might expect that the current directory is specified in a PWD environment variable.  Callers should explicitly set it themselves when invoking such programs.

or even just:

   Note: Callers are responsible for setting a PWD environment variable if necessary.

**Comment by bitdancer** [comment]
If you pass shell=True, PWD gests set, because it is a shell variable and the shell sets it.  If you don't, it doesn't, just like all the other shell-maintained variables.  I'm not sure it is worth calling out specifically in the Popen docs, myself; it is one consequence among \*many* of calling a program directly rather than through a shell.  (Note that if a shell is involved at any point, PWD gets set, so it is only binary programs that require that they be called from a shell that are at issue.)

Furthermore, it is not obvious that even all shells set PWD, although I believe that most do.

**Comment by 56441241-859c-4a4e-bf60-a8192e84927a** [comment]
Yes, but the Python docs have scary-looking warnings about using shell=True, so people (rightly) should avoid using shell=True if they don't think that they need it.  And in this case, people might not even know that they're invoking some binary that expects PWD from the shell.  

In my experience, it's not that uncommon for people to rewrite sh/bash shell scripts in Python.  Something starts off as a sh/bash script, evolves to a point where it's cumbersome to maintain, and it gets rewritten in Python.  But after it's rewritten, some invoked binaries might not behave as before, and it's unclear why.

When I encountered this problem myself, the first thing I tried was to explicitly set the `cwd` argument.  Some mention of the PWD environment variable would have saved me a lot of time.
